src/services/submissionService.ts

```diff
@@ -175,17 +176,38 @@ Promise<Result<gqlTypes.Submission>> => {
         const newSubmissionId = submissionRef.id
         const newSubmission = mapGqlEntityToDbEntity(submissionInput, newSubmissionId)
         
-        await submissionRef.set(newSubmission)
+        const batch = dbInstance.batch()
+        
+        batch.set(submissionRef, newSubmission)
+
+        const audit: dbSchema.AuditLog = {
+            actionType: gqlTypes.ActionType.Create,
+            objectType: gqlTypes.ObjectType.Submission,
+            schemaVersion: gqlTypes.SchemaVersion.V1,
+            updatedBy: 'client',
+            jsonData: JSON.stringify(newSubmission)
+        }
 
-        const createdSubmission = await getSubmissionById(newSubmissionId)
+        const createdAuditLog = await createAuditLog(audit)
 
-        // if we didn't get it back or have errors, this is an actual error.
-        if (createdSubmission.hasErrors || !createdSubmission.data) {
-            throw new Error(`${JSON.stringify(createdSubmission.errors)}`)
+        if (!createdAuditLog.isSuccesful) {
+            // I should add more specific errors
+            throw new Error('Error adding audit log')
         }
 
+        await batch.commit()
+
+        const createdSubmission = mapDbEntityTogqlEntity(newSubmission)
+
+        // const createdSubmission = await getSubmissionById(newSubmissionId)
+
```

Comment:
I think this is not necessary so for now I commented it out please correct me if I'm wrong

Reply:
this is a subjective approach to if you should return the saved object or fetch a new one. 

returning the same has a risk of any side effects that happened aren't picked up, but it saves a whole database query. (for example, maybe there's some other code running in an independent task on a database change)
also, if I remember, we don't get back the id of the created submission without a fetch (but it's been a while so I could be wrong)

Reply:
I have un commented this.
But we do create id as well, so we have access to it without getting the submission back. 
```
   const newSubmissionId = submissionRef.id
   const newSubmission = mapGqlEntityToDbEntity(submissionInput, newSubmissionId)
```